src/app/public/modules/auth-http-client/auth-interceptor.spec.ts

```diff
@@ -1,87 +1,78 @@
 //#region imports
 
 import {
-  HttpEventType,
   HttpHandler,
   HttpParams,
-  HttpRequest,
-  HttpSentEvent
+  HttpRequest
 } from '@angular/common/http';
 
+import {
+  TestBed
+} from '@angular/core/testing';
+
 import {
   Observable
 } from 'rxjs/Observable';
 
 import 'rxjs/add/observable/of';
 
 import {
-  SkyAuthTokenContextArgs
-} from '../auth-http/auth-token-context-args';
+  SkyAppConfig,
+  SkyAppRuntimeConfigParams
+} from '@skyux/config';
+
+import {
+  SkyAuthTokenContextArgs,
+  SkyAuthTokenProvider
+} from '../auth-http';
 
 import {
   SkyAuthInterceptor
 } from './auth-interceptor';
 
+import {
+  SKY_AUTH_DEFAULT_PERMISSION_SCOPE
+} from './auth-interceptor-default-permission-scope';
+
 import {
   SKY_AUTH_PARAM_AUTH,
   SKY_AUTH_PARAM_PERMISSION_SCOPE
 } from './auth-interceptor-params';
 
 //#endregion
 
-class MockHttpHandler extends HttpHandler {
-  public handle() {
-    return Observable.of({
-      type: HttpEventType.Sent
-    } as HttpSentEvent);
-  }
-}
+type Spy<T> = { [Method in keyof T]: jasmine.Spy; };
 
 describe('Auth interceptor', () => {
-  let mockTokenProvider: any;
-
-  function createInteceptor(envId?: string, leId?: string, url?: string) {
-    return new SkyAuthInterceptor(
-      mockTokenProvider,
-      {
-        runtime: {
-          params: {
-            get: (name: string) => {
-              switch (name) {
-                case 'envid':
-                  return envId;
-                case 'leid':
-                  return leId;
-                default:
-                  return undefined;
-              }
-            },
-            getUrl: () => url || 'https://example.com/get/'
-          }
-        } as any,
-        skyux: {}
-      });
-  }
-
-  function createRequest(params?: HttpParams) {
-    const request = new HttpRequest(
+  let mockTokenProvider: Spy<SkyAuthTokenProvider> = jasmine.createSpyObj('SkyAuthTokenProvider', ['getContextToken']);
```

Comment:
All of these variable assignments should be done in the `beforeEach` to make sure they are pristine for each test block.

```
describe('Auth interceptor', () => {
  let mockTokenProvider: Spy<SkyAuthTokenProvider>;
  // etc...

  beforeEach(() => {
    mockTokenProvider = jasmine.createSpyObj('SkyAuthTokenProvider', ['getContextToken']);
  });
```

Reference: https://jasmine.github.io/2.0/introduction.html#section-Spies:_%3Ccode%3EcreateSpyObj%3C/code%3E

Reply:
Fixed.